lib/phoenix/pubsub/local.ex

```diff
@@ -0,0 +1,212 @@
+defmodule Phoenix.PubSub.Local do
+  use GenServer
+
+  @moduledoc """
+  PubSub implementation for handling local-node process groups
+
+  This modules is used by Phoenix pubsub adapters to handle their
+  local node topic subscriptions. See `Phoenix.PubSub.PG2Adapter`
+  for an example integration.
+  """
+
+  @doc """
+  Starts the server
+
+    * `server_name` - The name to registered the server under
+
+  """
+  def start_link(server_name) do
```

Comment:
As we discussed, we should always expect a name, this will make it much easier to put it inside a supervision tree.